src/cdk-experimental/drag-drop/drag.ts

```diff
@@ -0,0 +1,493 @@
+  private _placeholder: HTMLElement;
+
+  /** Coordinates within the element at which the user picked up the element. */
+  private _pickupPositionInElement: Point;
+
+  /** Coordinates on the page at which the user picked up the element. */
+  private _pickupPositionOnPage: Point;
+
+  /** CSS `transform` applied to the element when it isn't being dragged.  */
+  private _passiveTransform: Point = {x: 0, y: 0};
+
+  /** CSS `transform` that is applied to the element while it's being dragged. */
+  private _activeTransform: Point = {x: 0, y: 0};
+
+  /** Whether the element is being dragged. */
+  private _isDragging = false;
+
+  /** Whether the element has been moved since the user started dragging it. */
+  private _hasMoved = false;
+
+  /** Drop container that the element was in at the time it was picked up. */
+  private _initialContainer: CdkDropContainer;
+
+  /** Element that can be used to drag the draggable item. */
+  @ContentChild(CdkDragHandle) _handle: CdkDragHandle;
+
+  /** Element that will be used as a template to create the draggable item's preview. */
+  @ContentChild(CdkDragPreview) _previewTemplate: CdkDragPreview;
+
+  /**
+   * Element that will be used as a template to create a placeholder
+   * for the draggable item while it's being dragged.
+   */
+  @ContentChild(CdkDragPlaceholder) _placeholderTemplate: CdkDragPlaceholder;
+
+  /** Emits when the user starts dragging the item. */
+  @Output() started = new EventEmitter<CdkDragStart>();
+
+  /** Emits when the user stops dragging an item in the container. */
+  @Output() ended = new EventEmitter<CdkDragEnd>();
+
+  /** Emits when the user has moved the item into a new container. */
+  @Output() entered = new EventEmitter<CdkDragEnter<any>>();
+
+  /** Emits when the user removes the item its container by dragging it into another container. */
+  @Output() exited = new EventEmitter<CdkDragExit<any>>();
+
+  /** Emits when the user drops the item inside a container. */
+  @Output() dropped = new EventEmitter<CdkDragDrop<any>>();
+
+  constructor(
+    /** Element that the draggable is attached to. */
+    public element: ElementRef<HTMLElement>,
+    /** Droppable container that the draggable is a part of. */
+    @Inject(CDK_DROP_CONTAINER) @Optional() @SkipSelf() public dropContainer: CdkDropContainer,
+    @Inject(DOCUMENT) document: any,
+    private _ngZone: NgZone,
+    private _viewContainerRef: ViewContainerRef) {
+      this._document = document;
+    }
+
+  /**
+   * Returns the element that is being used as a placeholder
+   * while the current element is being dragged.
+   */
+  getPlaceholderElement(): HTMLElement {
+    return this._placeholder;
+  }
+
+  ngAfterContentInit() {
+    // TODO: doesn't handle (pun intended) the handle being destroyed
+    const element = (this._handle ? this._handle.element : this.element).nativeElement;
+    element.addEventListener('mousedown', this._pointerDown);
```

Comment:
it looks like these event listeners are not cleaned up with a `removeEventListener`

Reply:
They're not, but presumably they live for the lifetime of the directive since they're bound either on the host or the handle which is inside the host.